distributed/client.py

```diff
@@ -262,7 +263,11 @@ def __str__(self):
     __repr__ = __str__
 
     def __await__(self):
-        return self._result().__await__()
+        tornado_future = self._result()
+        if isinstance(self.client.loop, tornado.platform.asyncio.BaseAsyncIOLoop):
```

Comment:
Given that both libraries have similar terms (ioloop, future, etc.) I've chosen to be fully explicit here.